gateway/app/schemas/qodProvisioning.py

```diff
@@ -0,0 +1,219 @@
+        description="REQUIRED. An access token is a previously acquired token granting access to the target resource.",
+    )
+    accessTokenExpiresUtc: datetime = Field(
+        description="REQUIRED. An absolute UTC instant at which the token shall be considered expired.",
+    )
+    accessTokenType: AccessTokenType = Field(
+        description="REQUIRED. Type of the access token (See [OAuth 2.0](https://tools.ietf.org/html/rfc6749#section-7.1)).",
+    )
+    refreshToken: str = Field(
+        description="REQUIRED. An refresh token credential used to acquire access tokens.",
+    )
+    refreshTokenEndpoint: AnyUrl = Field(
+        description="REQUIRED. A URL at which the refresh token can be traded for an access token.",
+    )
+
+
+class Type(Enum):
+    org_camaraproject_qod_provisioning_v0_status_changed = (
+        "org.camaraproject.qod-provisioning.v0.status-changed"
+    )
+
+
+class Specversion(Enum):
+    field_1_0 = "1.0"
+
+
+class Datacontenttype(Enum):
+    application_json = "application/json"
+
+
+class CloudEvent(BaseModel):
+    id: str = Field(
+        description="Identifier of this event, that must be unique in the source context.",
+    )
+    source: str = Field(
+        description="Identifies the context in which an event happened in the specific Provider Implementation.",
+    )
+    type: Type = Field(description="The type of the event.")
+    specversion: Specversion = Field(
+        description="Version of the specification to which this event conforms (must be 1.0 if it conforms to cloudevents 1.0.2 version)",
+    )
+    datacontenttype: Optional[Datacontenttype] = Field(
+        None,
+        description='media-type that describes the event payload encoding, must be "application/json" for CAMARA APIs',
+    )
+    data: Optional[Dict[str, Any]] = Field(
+        None,
+        description="Event notification details payload, which depends on the event type",
+    )
+    time: datetime = Field(
+        description="Timestamp of when the occurrence happened. It must follow RFC 3339\n",
+    )
+
+
+class StatusInfo(Enum):
+    NETWORK_TERMINATED = "NETWORK_TERMINATED"
+    DELETE_REQUESTED = "DELETE_REQUESTED"
+
+
+class Status(Enum):
+    REQUESTED = "REQUESTED"
+    AVAILABLE = "AVAILABLE"
+    UNAVAILABLE = "UNAVAILABLE"
+
+
+class StatusChanged(Enum):
+    AVAILABLE = "AVAILABLE"
+    UNAVAILABLE = "UNAVAILABLE"
+
+
+class Data(BaseModel):
+    provisioningId: ProvisioningId
+    status: Optional[StatusChanged] = None
+    statusInfo: Optional[StatusInfo] = None
+
+
+class EventStatusChanged(CloudEvent):
+    eventData: Data = Field(description="Event details depending on the event type")
+
+
+class DeviceIpv4Addr(BaseModel):
+    publicAddress: IPv4Address
+    privateAddress: Optional[IPv4Address] = None
+    publicPort: Optional[Port] = None
+
+    @model_validator(mode="after")
+    def any_of(cls, v: Any) -> Any:
+        if not v.privateAddress and not (v.publicAddress and v.publicPort):
```

Comment:
`publicAddress` will always be set so no need to check it in the condition. Also prefer using `is not None` instead of `not`.